check.go

```diff
@@ -268,6 +287,29 @@ func (c *CheckRunner) UpdateCheck(checkID types.CheckID, status, output string)
 		return
 	}
 
+	// Do nothing if update is idempotent
```

Comment:
I think the logic should be:
```go
if status == api.HealthPassing {
	check.passes++
	check.failures = 0
	// Early return if the status has changed to passing, but we haven't hit the
	// minimum .
	if check.Status != api.HealthPassing && check.passes < c.SuccessBeforePassing {
		return
	}
} else {
	check.failures++
	check.passes = 0
	// Early return if the status has changed to failing, but we haven't hit the
	// minimum.
	if check.Status == api.HealthPassing && check.failures < c.FailuresBeforeCritical {
		return
	}
}

// Do nothing if update is idempotent.
if check.Status == status && check.Output == output {
	return
}
```

Reply:
I simplified the logic, it's similar to your proposal, but I kept the idempotent case first to return faster. I don't have a strong opinion about it, but it "seemed" better. I preferred to set the counters in 2 places but having less `check.Status` checks... I'm open to discussion though.